blocks/graph.py

```diff
@@ -520,3 +521,82 @@ def apply_dropout(computation_graph, variables, drop_prob, rng=None,
         replacement.tag.replacement_of = variable
 
     return computation_graph.replace(replacements)
+
+
+def apply_batch_normalization(computation_graph, variables, gammas,
+                              betas, axis=0, epsilon=1e-7):
+    """Returns a graph to variables in a computational graph.
+
+    Parameters
+    ----------
+    computation_graph : instance of :class:`ComputationGraph`
+        The computation graph.
+    variables : list of :class:`~tensor.TensorVariable`
+        Variables to be batch-normalized.
+    gammas : list of :class:`~tensor.TensorVariable`
+        Scale coefficients for the normalized variables.
+    betas : list of :class:`~tensor.TensorVariable`
+        Shift coefficients for the normalized variables.
+    axis : int or iterable, optional
+        Batch axis, or batch axes. Defaults to 0.
+    epsilon : float, optional
+        Stabilization constant. Defaults to 1E-7.
+
+    Notes
+    -----
+    For more information, see [BN]_.
+
+    .. [BN] Sergey Ioffe and Christian Szegedy. *Batch Normalization:
+       Accelerating Deep Network Training by Reducing Internal Covariate
+       Shift*, arXiv:1502.03167.
+
+    Examples
+    --------
+    >>> from theano import function
```

Comment:
Doctests are executed in the same namespace as the file I think, so `tensor` is already available. I prefer not importing everything again in the docstring, because it turns docstrings into 50% import statements otherwise.